### PR TITLE
Refactor UUID generation

### PR DESCRIPTION
**Pull request by rsanchez-xtillion**
This PR adds the generate_uuid() UDF to replace all the uuid_string() calls when generating ids. The changes are:
 - Added file in utils directory with the new udf
 - Replaced all uuid_string calls with new udf
 - Passed corresponding columns as arguments to the udf for deterministic id generation

### CODE REVIEW

File: migration_original/ODS1Stage/functions/Utils/generate_uuid.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE OR REPLACE FUNCTION generate_uuid(name VARCHAR(16777216) DEFAULT NULL)
```
Comment (carresemata) [inline migration_original/ODS1Stage/functions/Utils/generate_uuid.sql:1]:
```suggestion
CREATE OR REPLACE FUNCTION ODS1_STAGE_TEAM.UTILS.generate_uuid(name VARCHAR(16777216) DEFAULT NULL)
```

Comment (carresemata) [reply migration_original/ODS1Stage/functions/Utils/generate_uuid.sql:1]: i think using another variable name is better than using name, maybe we can use code or something more descriptive

File: migration_original/ODS1Stage/tables/Base/ClientProductEntityToDisplayPartnerPhone/spu_translated_ClientProductEntityToDisplayPartnerPhone.sql [hunk 2]
```diff
@@ -687,7 +687,7 @@ select_statement_2:= $$     with cte_swimlane as (
     --- insert Statement
     insert_statement:= ' 
                 insert(clientproductentitytodisplaypartnerphoneid,ClientProductToEntityID,DisplayPartnerCode,PhoneTypeID,PhoneNumber,SourceCode,LastUpdateDate)
```
Comment (carresemata) [inline migration_original/ODS1Stage/tables/Base/ClientProductEntityToDisplayPartnerPhone/spu_translated_ClientProductEntityToDisplayPartnerPhone.sql:689]: can you modify the identation of this please so the inserted columsn have a line each

File: migration_original/ODS1Stage/tables/Base/ClientProductToCallCenter/spu_translated_ClientProductToCallCenter.sql [hunk 3]
```diff
@@ -49,7 +49,7 @@ select_statement := $$
 
 --- insert Statement
 insert_statement := ' insert (ClientProductToCallCenterID,ClientToProductID, CallCenterID, ActiveFlag, SourceCode, LastUpdateDate)
```
Comment (carresemata) [inline migration_original/ODS1Stage/tables/Base/ClientProductToCallCenter/spu_translated_ClientProductToCallCenter.sql:51]: please fix identation

File: migration_original/ODS1Stage/tables/Base/ClientProductToPartner/spu_translated_ClientProductToPartner.sql [hunk 4]
```diff
@@ -190,7 +190,7 @@ insert_statement := $$
                          )
                     values 
                         (
-                        uuid_string(),                        
+                        utils.generate_uuid(source.clienttoproductid || source.partnerid || source.sourcecode),
```
Comment (carresemata) [inline migration_original/ODS1Stage/tables/Base/ClientProductToPartner/spu_translated_ClientProductToPartner.sql:193]: are you sure sourcecode should be here? don't think so

File: migration_original/ODS1Stage/functions/Utils/generate_uuid.sql (hunk 1 above)
Comment (rsanchez-xtillion) [reply migration_original/ODS1Stage/functions/Utils/generate_uuid.sql]: I used name because that is the official name of the argument for version 5 UUIDs

Comment (rsanchez-xtillion) [reply migration_original/ODS1Stage/functions/Utils/generate_uuid.sql]: I did commit the change to the function name to ODS1_STAGE_TEAM.UTILS.generate_uuid

**Review by carresemata — CHANGES_REQUESTED** [review]

File: migration_original/ODS1Stage/tables/Base/ClientProductToPartner/spu_translated_ClientProductToPartner.sql (hunk 4 above)
Comment (rsanchez-xtillion) [reply migration_original/ODS1Stage/tables/Base/ClientProductToPartner/spu_translated_ClientProductToPartner.sql:193]: My bad, it shouldn't be there, I will remove it

File: migration_original/ODS1Stage/tables/Base/ClientProductEntityToDisplayPartnerPhone/spu_translated_ClientProductEntityToDisplayPartnerPhone.sql (hunk 2 above)
Comment (rsanchez-xtillion) [reply migration_original/ODS1Stage/tables/Base/ClientProductEntityToDisplayPartnerPhone/spu_translated_ClientProductEntityToDisplayPartnerPhone.sql:689]: Ok

File: migration_original/ODS1Stage/tables/Base/ClientProductToCallCenter/spu_translated_ClientProductToCallCenter.sql (hunk 3 above)
Comment (rsanchez-xtillion) [reply migration_original/ODS1Stage/tables/Base/ClientProductToCallCenter/spu_translated_ClientProductToCallCenter.sql:51]: Ok

File: migration_original/ODS1Stage/functions/Utils/generate_uuid.sql [hunk 5]
```diff
@@ -0,0 +1,13 @@
+CREATE OR REPLACE FUNCTION ODS1_STAGE_TEAM.UTILS.generate_uuid(name VARCHAR(16777216) DEFAULT NULL)
+RETURNS VARCHAR(16777216)
+LANGUAGE SQL
+AS
+$$
+CASE
+    WHEN name IS NOT NULL THEN
+        UUID_STRING('fe971b24-9572-4005-b22f-351e9c09274e', name)
```
Comment (carresemata) [inline migration_original/ODS1Stage/functions/Utils/generate_uuid.sql:8]: maybe add a comment explaining why you used this id and the fact that is random and the purpose of making the id generation stable

**Review by carresemata — CHANGES_REQUESTED** [review]

File: migration_original/ODS1Stage/functions/Utils/generate_uuid.sql (hunk 5 above)
Comment (rsanchez-xtillion) [reply migration_original/ODS1Stage/functions/Utils/generate_uuid.sql:11]: I added comments explaining the function

**Pull request merged**